zipline/algorithm.py

```diff
@@ -853,10 +850,20 @@ def calculate_capital_changes(self, dt, emission_rate, is_interday):
                       "('target' or 'delta')" % capital_change)
             return
 
-        self.capital_change_deltas.update({dt: capital_change_amount})
         self.perf_tracker.process_capital_change(capital_change_amount,
                                                  is_interday)
 
+        packet = {
+            'capital_change': [
+                {'date': dt,
+                 'type': 'cash',
+                 'target': capital_change['value'] if
+                 capital_change['type'] == 'target' else None,
```

Comment:
Since we already check these cases above, would it make sense to construct `packet` there instead of doing the same check for 'type' here?